### ISSUE TITLE
Conda package

### ISSUE THREAD

**Issue by Zethson** [saeyslab/nichenetr#95]
Hi,

would it be possible that you add nichenet to Bioconda?

Thank you very much

**Comment by browaeysrobin** [comment]
Hi @Zethson 

I will have a look into this.

**Comment by ejc043** [comment]
Is there any update on the conda implementation?

**Comment by browaeysrobin** [comment]
@ejc043 
Because of multiple projects and lack of time, this is currently not one of my main priorities (this can change if you can convince me that adding it to Bioconda could be very beneficial).

**Comment by akiss-me** [comment]
> @ejc043 Because of multiple projects and lack of time, this is currently not one of my main priorities (this can change if you can convince me that adding it to Bioconda could be very beneficial).

Today I spent 5 hours on the installation of nichenet, some of the dependencies outdated, conda could help save time for installation.

**Comment by akiss-me** [comment]
This is the code for quick nichenet installation using conda:

conda create -y --name nichenetr_env r=3.6.3 \
&& conda activate nichenetr_env && conda update --all \
        && conda install -y -c conda-forge -c bioconda -c defaults \
                        r-dplyr \
                        r-tidyr \
                        r-igraph \
                        r-matrix \
                        r-fdrtool \
                        r-rocr \
                        r-catools \
                        r-data.table \
                        bioconductor-limma \
                        r-readr \
                        r-hmisc \
                        r-tibble \
                        r-caret \
                        r-purrr \
                        r-randomforest \
                        r-diagrammer \
                        r-ggplot2 \
                        r-mlrmbo \
                        r-parallelmap \
                        r-emoa \
                        r-dicekriging \
                        r-e1071 \
                        r-seurat \
                        r-cowplot \
                        r-ggpubr \
                        r-magrittr \
                        r-circlize \
                        bioconductor-complexheatmap \
                        r-knitr \
                        r-rmarkdown \
                        r-testthat \
                        r-domc \
                        r-covr \
                        r-tidyverse \
                        r-devtools \
                        r-remotes \
                        r-pbkrtest \
                        r-lme4 \
                        r-car \
                        r-rgeos \
                        r-parallelmap \
                        r-mlr \
                        r-farver \
                        r-fastmap \
                        r-bit \
                        r-bit64 \
                        r-tzdb \
                        r-vroom \
                        r-parallelly \
                        r-progressr \
                        r-hardhat \
                        r-pillar \
                        r-cli \
                        r-cpp11 \ 
                        r-glue \
                        r-rlang \
                        r-tibble \
                        r-vctrs \
                        r-spatstat=1.64_1 \
                && conda clean -a -y \
        && R -e 'remotes::install_github("saeyslab/nichenetr", upgrade = "never"); q()'

**Comment by sfortma2** [comment]
> > @ejc043 Because of multiple projects and lack of time, this is currently not one of my main priorities (this can change if you can convince me that adding it to Bioconda could be very beneficial).
> 
> Today I spent 5 hours on the installation of nichenet, conda could help save time for installation.

Yup, same here. Took me 2 days of on and off attempts to get a working conda environment for NicheNet. Love the software though!

**Comment by shufanzhang** [comment]
It also take me half day to install this package in micromamba, R4.0.0 can not install successful for that some package rely on R>4.1. My R version is  4.2.3. Here is my successful install：

**#build a clean environment:**
micromamba create -y -n CCI
micromamba activate CCI

**#install R-base and packages that can not be installed by devtools::install_github:**
micromamba install -c bioconda -c conda-forge  r-base r-devtools \
bioconductor-limma r-hmisc r-randomforest r-mlrmbo r-ggpubr bioconductor-complexheatmap r-ggtern

**#run this in R:**
devtools::install_github("saeyslab/nichenetr")

**Comment by Elo-mars** [comment]
> It also take me half day to install this package in micromamba, R4.0.0 can not install successful for that some package rely on R>4.1. My R version is 4.2.3. Here is my successful install：
> 
> **#build a clean environment:** micromamba create -y -n CCI micromamba activate CCI
> 
> **#install R-base and packages that can not be installed by devtools::install_github:** micromamba install -c bioconda -c conda-forge r-base r-devtools bioconductor-limma r-hmisc r-randomforest r-mlrmbo r-ggpubr bioconductor-complexheatmap
> 
> **#run this in R:** devtools::install_github("saeyslab/nichenetr")


For conda users, and a bit of an update from November 2023 (and after 4 hours??)

this works for me:

`conda create --name MN python=3.12
conda activate MN

conda install -c conda-forge r-base
conda install -c conda-forge r-seurat
conda install -c bioconda bioconductor-singlecellexperiment
conda install -c conda-forge r-dplyr
conda install -c conda-forge r-ggplot2
conda install -c bioconda -c conda-forge r-base r-devtools
conda install  bioconductor-limma r-hmisc r-randomforest r-mlrmbo r-ggpubr bioconductor-complexheatmap
conda install -c conda-forge r-caret
conda install -c conda-forge r-ggforce
# conda install -c bioconda r-seurat
conda install -c conda-forge r-rspectra r-rcpphnsw


R

devtools::install_github("saeyslab/nichenetr") # with warnings
q()
n

conda install -c bioconda bioconductor-muscat bioconductor-scater bioconductor-edger bioconductor-sva bioconductor-scran
conda install -c conda-forge r-ggraph


devtools::install_github("saeyslab/multinichenetr")
`


but thanks for the package, it's wonderful :)
